src/relay/backend/aot_executor_codegen.cc

```diff
@@ -215,11 +217,21 @@ class AOTOnDemandAllocator : public transform::DeviceAwareExprVisitor {
    * \return The corresponding token.
    */
   StorageInfo GetStorage(const Expr& expr) {
+    Expr true_expr = expr;
+
+    // Don't get storage for let nodes.
+    while (const auto* let_node = true_expr.as<LetNode>()) {
```

Comment:
Won't we achieve the same effect by using something from here ? : 
https://github.com/apache/tvm/blob/f0854ae456a98581d7cc4004fc848f73879e5c3e/src/relay/transforms/device_aware_visitors.h#L247-L258

Reply:
hmm I can't see a way to make use of these here, perhaps I've missed something?

Reply:
I was thinking instead of : 

```
void PreVisitLetBinding_(const Var& var, const Expr& value) final {
    LOG(FATAL) << "let is not supported.";
    VisitExpr(value);
    StorageInfo si = GetStorage(value);
    storage_device_map_[var] = si;
  }
```

if we do PostVisitLet, wont we have the same effect (i.e. the body is visited)?

Reply:
No objection from me if you want to just do the loop directly as I find that much clearer. In my travels I found switching to pre- and post-let-binding visitor overloads made the most sense when you find you need to accumulate intermediate rewritten bindings on a local stack in order to reconstruct the final let chain. ExpandANormalForm also achieves the same effect all be it by relying on the implicit memoization to capture the intermediate results (which hurts my head every time I come across it!)

Reply:
Me and @manupa-arm discussed this offline and it turns out we can just avoid let's being used in `GetStorage`